test/system/src/test/java/io/pravega/test/system/AbstractReadWriteTest.java

```diff
@@ -552,6 +573,89 @@ private EventWriterConfig buildWriterConfig() {
+            String value = convertStringToBinary("Hello World");
+            Integer keyId = i;
+            CompletableFuture.runAsync(() -> {
+                try {
+                    KeyValueTableInfo kvt = new KeyValueTableInfo(scope, kvtName);
+                    keyValueTable = keyValueTableFactory.forKeyValueTable(kvt.getKeyValueTableName(), KEY_SERIALIZER, VALUE_SERIALIZER, KeyValueTableClientConfiguration.builder().build());
+                    log.info("Start creating KVP id:{} in KVT {}", keyId, kvtName);
+                    // Create KVP
+                    Version insertKvp = keyValueTable.put(null, keyId, value).join();
+                    // get KVP and verify the insert value length
+                    Assert.assertEquals("Value length is not matching", value.length(),
+                            keyValueTable.get(null, keyId).join().getValue().length());
+                    // Call API to update kvp table entry
+                    updateKVPEntry(scope, kvtName, keyId, keyValueTableFactory);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }, Executors.newFixedThreadPool(1));
+        }
+    }
+
+    void updateKVPEntry(String scope, String kvtName, int keyId, KeyValueTableFactory keyValueTableFactory) {
+        log.info("Scope name {} KVT name {} KVP name {}", scope, kvtName, keyId);
+        KeyValueTableInfo kvt = new KeyValueTableInfo(scope, kvtName);
+        keyValueTable = keyValueTableFactory.forKeyValueTable(kvt.getKeyValueTableName(), KEY_SERIALIZER, VALUE_SERIALIZER, KeyValueTableClientConfiguration.builder().build());
+        String replaceValue = "";
+        for (int i = 1; i < 10000; i++) {
+            replaceValue = replaceValue + convertStringToBinary("Pravega");
+            Version keyVersion = keyValueTable.get(null, keyId).join().getKey().getVersion();
+            Version replaceKvp = keyValueTable.replace(null, keyId, replaceValue, keyVersion).join();
+            long keyValueLength = keyValueTable.get(null, keyId).join().getValue().length();
+            Assert.assertEquals("Value length is not matching", replaceValue.length(), keyValueLength);
+            log.info("For KVP {} number of update value {} lenght size {}", keyId, i, keyValueLength);
+            kvpUpdateCount = kvpUpdateCount + 1;
+            log.info("KVP update count {}", kvpUpdateCount);
+        }
+    }
+
+    private static class IntegerSerializer implements Serializer<Integer> {
+        public ByteBuffer serialize(Integer value) {
+            return ByteBuffer.allocate(Integer.BYTES).putInt(0, value);
+        }
+
+        public Integer deserialize(ByteBuffer serializedValue) {
+            return serializedValue.getInt();
+        }
+    }
+
+    public static String convertStringToBinary(String input) {
+        StringBuilder result = new StringBuilder();
+        char[] chars = input.toCharArray();
+        for (char aChar : chars) {
```

Comment:
This looks expensive, isn't there any other way to do that? Also, why are you executing `replaceAll(" ", "0"))` on every iteration?

Reply:
remove `convertStringToBinary` from loop and calling only twice in the execution, Also removed `replaceAll(" ", "0"))` from every iteration and add in return result output